File: front-end/src/application/service/accessService.ts
import { AccessRepository } from "../repository/accessRepository";
import { Access } from "../../domain/entity/access";
import { Either } from "../../utils/ull-either";
import { DataError } from "../../utils/ull-data-error";

export class AccessService {
    private accessRepository: AccessRepository;

    constructor(accessRepository: AccessRepository) {
        this.accessRepository = accessRepository;
    }
    
    public save = (newAccess: Access): Promise<Either<DataError, Access>> => {
        return this.accessRepository.save(newAccess);
    }

    public findAll = (pageRequest: any): Promise<Either<DataError, Access[]>> => {
        return this.accessRepository.findAll(pageRequest);
    }

    public deleteAll = (): Promise<Either<DataError, Access[]>> => {
        return this.accessRepository.deleteAll();
    }
}

File: back-end/src/main/java/com/trucks/application/service/AccessService.java
package com.trucks.application.service;

import java.util.List;

import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.data.domain.Page;
import org.springframework.data.domain.PageRequest;

import com.trucks.application.port.AccessRepository;
import com.trucks.domain.entity.Access;

public class AccessService {

    @Autowired
    AccessRepository accessRepository;

    public Page<Access> findAll(PageRequest page) {
        return this.accessRepository.findAll(page);
    }

    public Access save(Access newAccess) {
        return this.accessRepository.save(newAccess);
    }

    public List<Access> deleteAll() {
        return this.accessRepository.deleteAll();
    }
}
